src/App.tsx

```diff
@@ -1,42 +1,34 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
+import { Tabs } from './components/Tabs';
+import { Tab } from './types/Tab';
 
-export const tabs = [
+export const tabsFromServer = [
   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
 ];
 
 export const App: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState(tabsFromServer[0]);
+
+  const selectNewTab = (tab: Tab) => {
+    setSelectedTab(tab);
+  };
+
   return (
     <div className="section">
       <h1 className="title">
-        Selected tab is Tab 1
+        {`Selected tab is ${selectedTab.title}`}
       </h1>
 
-      <div data-cy="TabsComponent">
-        <div className="tabs is-boxed">
-          <ul>
-            <li className="is-active" data-cy="Tab">
-              <a href="#tab-1" data-cy="TabLink">Tab 1</a>
-            </li>
-
-            <li data-cy="Tab">
-              <a href="#tab-2" data-cy="TabLink">Tab 2</a>
-            </li>
-
-            <li data-cy="Tab">
-              <a href="#tab-3" data-cy="TabLink">Tab 3</a>
-            </li>
-          </ul>
-        </div>
-
-        <div className="block" data-cy="TabContent">
-          Some text 1
-        </div>
-      </div>
+      <Tabs
+        tabs={tabsFromServer}
+        selectedTabId={selectedTab.id}
```

Comment:
Look, you have the whole tab in a state, but pass only selectedTab.id to the Tabs component. And after that, you find a selected tab there. Why don't you pass the whole tab at once

Reply:
I didn't fully understand exactly what you meant, but if I understood correctly.... it gave me an idea of how to write it differently and I revised the logic in general😅

Reply:
@Anastasiia-Svintsova  I have already understood what you mean....I passed the whole object(not just "id")... and it works....but somehow 4 tests failed...maybe the problem in the tests themselves? maybe they expect to accept "id" instead of "object"?